lib/OpenQA/WebAPI/Controller/API/V1.pm

```diff
@@ -53,7 +53,7 @@ sub auth {
                 $user = $api_key->user;
             }
             else {
-                $self->app->log->error("hmac check failed");
+                $self->app->log->error(sprintf "Wrong api secret for key '%s' (belongs to user '%s') used.", $api_key->key, $api_key->user->username);
```

Comment:
Please also log the ip address, and keep in mind that we might be behind a proxy :)